aws/rust-runtime/aws-config/src/profile/parser/section.rs

```diff
@@ -0,0 +1,314 @@
+use crate::profile::parser::parse::to_ascii_lowercase;
+use std::collections::HashMap;
+
+/// Key-Value property pair
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Property {
+    key: String,
+    value: String,
+}
+
+impl Property {
+    /// Value of this property
+    pub fn value(&self) -> &str {
+        &self.value
+    }
+
+    /// Name of this property
+    pub fn key(&self) -> &str {
+        &self.key
+    }
+
+    /// Creates a new property
+    pub fn new(key: String, value: String) -> Self {
+        Property { key, value }
+    }
+}
+
+type SectionKey = String;
+type SectionName = String;
+type SubPropertyGroupName = String;
+type SubPropertyKey = String;
+type SubPropertyValue = String;
+
+#[allow(clippy::type_complexity)]
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct Properties {
+    // [section-key section-name]
+    // sub-property-group-name =
+    //   sub-property-name = sub-property-value
+    inner: HashMap<
+        SectionKey,
+        HashMap<
+            SectionName,
+            HashMap<SubPropertyGroupName, HashMap<SubPropertyKey, SubPropertyValue>>,
+        >,
+    >,
+}
+
+impl Properties {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn insert_at(&mut self, path: &[&str], value: String) {
+        if path.is_empty() {
+            return;
+        }
+
+        if path.len() != 4 {
```

Comment:
Doesn't it also need to support 3 for the non-nested properties that are needed from the active profile section?

Reply:
I updated this to work with a simpler key type.